src/main/java/chess/service/GameService.java

```diff
@@ -0,0 +1,135 @@
+import chess.repository.dao.GameDao;
+import chess.repository.dao.PieceDao;
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Optional;
+import java.util.function.Supplier;
+
+public class GameService {
+
+    private final GameDao gameDao;
+    private final BoardDao boardDao;
+    private final PieceDao pieceDao;
+    private final TransactionManager transactionManager;
+
+    public GameService(GameDao gameDao, BoardDao boardDao, PieceDao pieceDao, TransactionManager transactionManager) {
+        this.gameDao = gameDao;
+        this.boardDao = boardDao;
+        this.pieceDao = pieceDao;
+        this.transactionManager = transactionManager;
+    }
+
+    public Optional<ChessGame> loadGame() {
+        int lastGameId = transactionManager.getData(gameDao::findLastGameId);
+        Optional<Board> board = transactionManager.getData(
+                connection -> createBoard(connection, lastGameId)
+        );
+        if (board.isEmpty()) {
+            return Optional.empty();
+        }
+
+        Optional<Color> savedTurn = transactionManager.getData(
+                connection -> gameDao.findGameById(connection, lastGameId)
+        );
+        if (savedTurn.isEmpty()) {
+            throw new IllegalStateException("DB에 저장된 턴이 없습니다.");
+        }
+
+        return Optional.of(new PlayingGame(lastGameId, board.get(), savedTurn.get()));
+    }
+
+    private Optional<Board> createBoard(Connection connection, int lastGameId) throws SQLException {
+        Map<Location, Integer> pieceIdsWithLocation = boardDao.findBoardById(connection, lastGameId);
+        if (pieceIdsWithLocation.isEmpty()) {
+            return Optional.empty();
+        }
+        Board board = generateBoard(connection, pieceIdsWithLocation);
+        return Optional.of(board);
+    }
+
+    private Board generateBoard(Connection connection, Map<Location, Integer> pieceLocations)
+            throws SQLException {
+        Map<Location, Piece> piecesWithLocation = new HashMap<>();
+        for (Entry<Location, Integer> locationIntegerEntry : pieceLocations.entrySet()) {
+            Location location = locationIntegerEntry.getKey();
+            Integer pieceId = locationIntegerEntry.getValue();
+            Piece piece = pieceDao.findPieceById(connection, pieceId);
+            piecesWithLocation.put(location, piece);
+        }
+        return new Board(piecesWithLocation);
+    }
+
+    public ChessGame createNewGame() {
+        int lastGameId = transactionManager.getData(gameDao::findLastGameId);
+        //TODO 방을 여러개 관리하게 되면 이전 데이터를 남겨두고 새로 만들어야 함
+        transactionManager.executeTransaction(
+                connection -> deleteGame(connection, lastGameId)
+        );
+
+        int newGameId = lastGameId;
+        if (newGameId == 0) {
+            newGameId = 1;
+        }
+        return new InitialGame(newGameId);
```

Comment:
```suggestion
        
        if (lastGameId == 0) {
            return new InitialGame(1);
        }
        return new InitialGame(lastGameId);
```
어떤 로직인지는 의도가 잘안보이는 로직이네요🤔

Reply:
헉, 의도가 훨씬 잘 드러나네요!!
제안 주신 코드로 변경 후 상수 추출을 하여 의도를 조금 더 잘 파악할 수 있도록 변경하겠습니다!